src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -11642,6 +11714,7 @@ void Assembler::evex_prefix(bool vex_r, bool vex_b, bool vex_x, bool evex_r, boo
 }
 
 void Assembler::vex_prefix(Address adr, int nds_enc, int xreg_enc, VexSimdPrefix pre, VexOpcode opc, InstructionAttr *attributes) {
+  bool is_extended = adr.base_needs_rex2() || adr.index_needs_rex2() || nds_enc >= 16 || xreg_enc >= 16;
```

Comment:
We could add an assert here:
if  (adr.base_needs_rex2() || adr.index_needs_rex2()) {
  assert(UseAPX, "APX features not enabled");
}

Reply:
Thanks, done.